src/Controller/UserActivateController.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Controller;
+
+use App\Exceptions\ExternalUserCodeException;
+use App\Service\ExternalUserService;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Attribute\AsController;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+#[AsController]
+class UserActivateController extends AbstractController
+{
+    public function __construct(
+        private readonly ExternalUserService $externalUserService,
+    ) {}
+
+    public function __invoke(Request $request): JsonResponse
+    {
+        $body = json_decode($request->getContent());
+        $activationCode = $body->activationCode;
+
+        try {
+            $this->externalUserService->activateExternalUser($activationCode);
+        } catch (ExternalUserCodeException $e) {
+            throw new HttpException($e->getCode(), $e->getMessage());
```

Comment:
On the fence on the exception here. The code given is important for the API client to understand the error. Is the the controller or the `ExternalUserService` who should choose the proper code? Doing it this way pushes the responsibility for choosing the correct HTTP code down to the service. But I think it's a bit of a code smell to have the `ExternalUserService`be required to know HTTP semantics.

Reply:
Fixed